hal/src/b5som/network/quectel_ncp_client.cpp

```diff
@@ -117,12 +124,9 @@ int QuectelNcpClient::init(const NcpClientConfig& conf) {
     // Initialize serial stream
     auto sconf = SERIAL_8N1 | SERIAL_FLOW_CONTROL_RTS_CTS;
 
-    // Hardware version
-    // V003 - 0x00 (disable hwfc)
-    // V004 - 0x01 (enable hwfc)
-    uint32_t hwVersion = 0xFF;
-    hal_get_device_hw_version(&hwVersion, nullptr);
-    if (hwVersion == 0x00) {
+    uint32_t hwVersion = HW_VERSION_UNDEFINED;
+    CHECK(hal_get_device_hw_version(&hwVersion, nullptr));
```

Comment:
`CHECK()` will return from function in case of failure, we should probably just resort to the default behavior.

Reply:
That's why I didn't add `CHECK` but using a default value. But I agree with your idea that adding `CHECK` here, if we're using B5SoM_V003 and reading flash data failed for some reason, we shouldn't enable HW flow control.

Reply:
Or we can combine these two conditions to not using HWFC

Reply:
We should, that is the point. The default behavior should be with flow control. Only if the revision is 100% the affected one we should enable it.

Reply:
We can simply remove `CHECK()` here?

Reply:
Yes, but we do need to check the return code from `hal_get_device_hw_version()`. So the condition for disabling for flow control becomes:
`if (ret == SYSTEM_ERROR_NONE && hwVersion == HAL_VERSION_B5SOM_V003)`

Reply:
Fixed